### PR TITLE
Add Config for web applications

### PR DESCRIPTION
**Pull request by PierreCookie**
Fix #9926

### CODE REVIEW

File: doc/oauth_setup.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Oauth Setup
+
+If you want to let party watch their own data, you will have to follow this guide
```
Comment (jmousset) [inline doc/oauth_setup.md]: if you want to set up a b2c portal ?

File: doc/oauth_setup.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+Those are used to delegate user connection.
+
+### Google account
+
+First of all you have to go [there](https://console.developers.google.com)
+
+If you don't have any project yet, then google will ask you to create one.
+
+Name it, choose your organization, your zone and wait until he is created.
```
Comment (jmousset) [inline doc/oauth_setup.md]: it is created

File: doc/oauth_setup.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+Take a look into first step section.
+
+You will see a subsection "Activate API ...", click it, you will be redirected.
+
+You see some APIs available but maybe not the one we need "GOOGLE+ API",
+so click on browse all and just type "GOOGLE+ API" in the search bar.
+
+Click on activate API
+
+Then a new API dashboard appear and ask you to create identifiers to start. So click it.
```
Comment (jmousset) [inline doc/oauth_setup.md]: appears

File: doc/oauth_setup.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+Which data would you acces ?
+Data user
+
+Then click on the button
+
+#### Second step
+
+Name your client id
+
+JS Origin ahorized :
```
Comment (jmousset) [inline doc/oauth_setup.md]: ahorized ?

File: doc/oauth_setup.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+Redirect URL authorized :
+
+Enter redirect address url like this : <http://path:port><API_PATH>/grant/connect/google/callback
+
+Click ont he button
+
+#### Third step
+
+Select your address mail and set the name that will be displayed for your user.
```
Comment (jmousset) [inline doc/oauth_setup.md]: mail address

File: doc/oauth_setup.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+#### Forth step
+
+Download the file, open it and open coog configuration
+
+``` bash
+./conf edit
+```
+
+Then just type those two environment
```
Comment (jmousset) [inline doc/oauth_setup.md]: variables

File: doc/oauth_setup.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+Click on the notification about your created project.
+
+You will be redirected into your dashboard project.
+
+Take a look into first step section.
+
+You will see a subsection "Activate API ...", click it, you will be redirected.
+
+You see some APIs available but maybe not the one we need "GOOGLE+ API",
```
Comment (jmousset) [inline doc/oauth_setup.md]: available APIs

File: config [hunk 8]
```diff
@@ -121,6 +121,11 @@ COOG_FONT_DIR=/usr/local/fonts          # folder to look in for fonts
 #-------------------------------------------------------------------------------#
 
 WEB_IMAGE=coopengo/web:master           # docker image
+COOG_API_URL=                           # server url
```
Comment (manalaboutayeb) [inline config]: Pas de valeurs par défaut?

File: doc/oauth_setup.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+Those are used to delegate user connection.
+
+### Google account
+
+First of all you have to go [there](https://console.developers.google.com)
+
+If you don't have any project yet, then google will ask you to create one.
+
+Name it, choose your organization, your zone and wait until he it is created.
```
Comment (manalaboutayeb) [inline doc/oauth_setup.md]: it is*

File: doc/oauth_setup.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+Take a look into first step section.
+
+You will see a subsection "Activate API ...", click it, you will be redirected.
+
+You see some available APIs but maybe not the one we need "GOOGLE+ API",
+so click on browse all and just type "GOOGLE+ API" in the search bar.
+
+Click on activate API
+
+Then a new API dashboard appears and ask you to create identifiers to start. So click it.
```
Comment (manalaboutayeb) [inline doc/oauth_setup.md]: asks

File: doc/oauth_setup.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+#### First step
+
+Which API do you use ?
+GOOGLE+ API
+
+Which plateform do you use to call API ?
+Web server (example: Node.js or Tomcat)
+
+Which data would you acces ?
```
Comment (manalaboutayeb) [inline doc/oauth_setup.md]: access*

File: doc/oauth_setup.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+JS Origin authorized :
+
+Enter your server address : <http://path:port>
+
+Redirect URL authorized :
+
+Enter redirect address url like this : <http://path:port><API_PATH>/grant/connect/google/callback
+
+Click ont he button
```
Comment (manalaboutayeb) [inline doc/oauth_setup.md]: on the*

File: doc/oauth_setup.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+#### Third step
+
+Select your mail address and set the name that will be displayed for your user.
+
+(optional) You can customize other things like CCG, Logo, HomePage and Privacy rules.
+
+#### Forth step
```
Comment (manalaboutayeb) [inline doc/oauth_setup.md]: fourth*

File: config (hunk 8 above)
Comment (PierreCookie) [reply config]: Nope c'est gérer coté api

**Pull request merged**